tests/test_ensemble.py

```diff
@@ -24,15 +24,40 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+import itertools
+import typing as t
+
 import pytest
 
 from smartsim.entity import _mock
 from smartsim.entity._new_ensemble import Ensemble
+from smartsim.entity.param_data_class import ParamSet
+from smartsim.error import errors
 
 pytestmark = pytest.mark.group_a
 
 _2x2_PARAMS = {"SPAM": ["a", "b"], "EGGS": ["c", "d"]}
-_2_PERM_STRAT = lambda p, n: [{"SPAM": "a", "EGGS": "b"}, {"SPAM": "c", "EGGS": "d"}]
+_2x2_EXE_ARG = {"EXE": [["a"], ["b", "c"]], "ARGS": [["d"], ["e", "f"]]}
+
+
+def user_created_function(
+    file_params: t.Mapping[str, t.Sequence[str]],
+    exe_arg_params: t.Mapping[str, t.Sequence[t.Sequence[str]]],
+    _n_permutations: int = 0,
+) -> list[ParamSet]:
+    # Create dictionaries for each parameter permutation
+    param_zip = [dict(zip(file_params, permutation)) for permutation in file_params][
+        :_n_permutations
+    ]
```

Comment:
did you mean
```suggestion
    param_zip = [dict(zip(file_params, permutation)) for permutation in file_params.values()][
        :_n_permutations
    ]
```
?

Not really a huge deal because its testing code, but it seemed off...